### PR TITLE
Add more \l commands.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add `\lc [PATTERN]` command for listing casts, optionally filtered by
affected type names using the specified regexp pattern.
Add `\lcI [PATTERN]` command for case-sensitive pattern matching.

Add `\lt [PATTERN]` command for listing object types, optionally filtered by
name using the specified regexp pattern.
Add `\ltI [PATTERN]` command for case-sensitive pattern matching.

Issue: #179

### CODE REVIEW

File: edb/repl/__init__.py [hunk 1]
```diff
@@ -435,6 +435,151 @@ def _list_scalar_types(
...
+                            width=1, align='left'),
+                        table.ColumnSpec(
+                            field='kind', title='Type of Cast',
+                            width=1, align='left'),
+                        table.ColumnSpec(
+                            field='volatility', title='Volatility',
+                            width=1, align='left'),
+                    ],
+                    data=result,
+                    max_width=min(max_width, 120),
```
Comment (1st1) [inline edb/repl/__init__.py]: Please move 120 into a constant (MAX_TABLE_WIDTH or something)

File: edb/repl/utils.py [hunk 2]
```diff
@@ -127,16 +127,21 @@ def normalize_name(name: str) -> str:
...
         flag = f'(?{flag})'
 
     qkw = {}
 
     if pattern:
-        qkw[f're_{field}'] = flag + pattern
-        return f'FILTER re_test(<str>$`re_{field}`, {field})', qkw
+        # the pattern is the same for all fields, so we only use one
+        # variable
+        qkw[f're_{fields[0]}'] = flag + pattern
```
Comment (1st1) [inline edb/repl/utils.py]: Please replace `f're_{fields[0]}'` with `'re_filter'`. This business with using the first element of the `fields` mapping for a parameter name is extremely confusing.

File: edb/repl/utils.py [hunk 3]
```diff
@@ -127,16 +127,21 @@ def normalize_name(name: str) -> str:
...
+        result = []
+        for field in fields:
+            result.append(f're_test(<str>$`re_{fields[0]}`, {field})')
+        return f'FILTER {" OR ".join(result)}', qkw
 
     else:
         # no FILTER clause necessary
```
Comment (1st1) [inline edb/repl/utils.py]: `return '', {}`

File: edb/repl/utils.py [hunk 4]
```diff
@@ -127,16 +127,21 @@ def normalize_name(name: str) -> str:
...
 
     qkw = {}
 
     if pattern:
-        qkw[f're_{field}'] = flag + pattern
-        return f'FILTER re_test(<str>$`re_{field}`, {field})', qkw
+        # the pattern is the same for all fields, so we only use one
+        # variable
+        qkw[f're_{fields[0]}'] = flag + pattern
+        result = []
```
Comment (1st1) [inline edb/repl/utils.py]:
```suggestion
        filters = []
```

File: edb/repl/__init__.py (hunk 1 above)
Comment (vpetrovykh) [reply edb/repl/__init__.py]: It's also used for other output formatting, so I'll just call it `MAX_WIDTH` and add a short comment.

File: edb/repl/utils.py (hunk 2 above)
Comment (vpetrovykh) [reply edb/repl/utils.py]: Yeah, at this point that's probably better. Originally I just thought that it made sense to correlate the name of the regexp parameter with the name of the prop it's applied to, but with this update it's no longer the case.

File: edb/repl/utils.py (hunk 3 above)
Comment (vpetrovykh) [reply edb/repl/utils.py]: OK

File: edb/repl/utils.py (hunk 4 above)
Comment (vpetrovykh) [reply edb/repl/utils.py]: Done

**Pull request merged**